crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -899,14 +900,13 @@ impl HolochainP2pActor {
         space: &DynSpace,
         loc: u32,
     ) -> HolochainP2pResult<Vec<(AgentPubKey, Url)>> {
-        let agent_list = space.peer_store().get_near_location(loc, 1024).await?;
-        let local_agents = space
-            .local_agent_store()
-            .get_all()
-            .await?
-            .into_iter()
-            .map(|a| a.agent().clone())
-            .collect::<Vec<_>>();
+        let agent_list = get_remote_agents_near_location(
```

Comment:
This was already being filtered for local agents, used common function to make it extra clear